cli/wallet/nep17.go

```diff
@@ -203,27 +206,86 @@ func getNEP17Balance(ctx *cli.Context) error {
 				tokenName = token.Name
 				tokenSymbol = token.Symbol
 				tokenDecimals = int(token.Decimals)
+				tokenFound = true
 			} else {
 				if name != "" {
 					continue
 				}
 				tokenSymbol = "UNKNOWN"
 			}
-			fmt.Fprintf(ctx.App.Writer, "%s: %s (%s)\n", tokenSymbol, tokenName, asset.StringLE())
-			amount := balances.Balances[i].Amount
-			if tokenDecimals != 0 {
-				b, ok := new(big.Int).SetString(amount, 10)
-				if ok {
-					amount = fixedn.ToString(b, tokenDecimals)
+			printAssetBalance(ctx, asset, tokenName, tokenSymbol, tokenDecimals, balances.Balances[i])
+		}
+		if name == "" || tokenFound {
+			continue
+		}
+		// Token was explicitly specified, but was not found among balances, thus either balance is 0
+		// or the token doesn't exists. Try to find token by its address/hash/name/symbol and print zero
+		// balance if found.
+		asset, err := flags.ParseAddress(name)
```

Comment:
Why don't you `getMatchingToken` here? It can be specified by name and be present in the wallet metadata.

Reply:
Fixed.